src/lib/AutoControlledComponent.js

```diff
@@ -120,10 +131,24 @@ export default class AutoControlledComponent extends Component {
 
   componentWillReceiveProps(nextProps) {
     if (super.componentWillReceiveProps) super.componentWillReceiveProps(nextProps)
+    const { autoControlledProps } = this.constructor
 
-    // props always win, update state with all auto controlled prop
-    const newState = _.pick(nextProps, this.constructor.autoControlledProps)
-    if (!_.isEmpty(newState)) this.setState(newState)
+    // Solve the next state for autoControlledProps
+    const newState = _.transform(autoControlledProps, (acc, prop) => {
+      const isNextUndefined = _.isUndefined(nextProps[prop])
+      const propWasRemoved = _.has(this.props, prop) && isNextUndefined
```

Comment:
Indeed, I've updated this and similar issues in `getAutoControlledStateValue()`.  I've also added several tests to ensure defaults are properly handled in both instances:

```
initial state
  ✔ uses the default prop is the regular prop is undefined
  ✔ uses the regular prop when a default is also defined

changing props
  ✔ does not return state to default props when setting props undefined
```